src/OpenTelemetry.Abstractions/Trace/LinksProvider.cs

```diff
@@ -0,0 +1,36 @@
+// <copyright file="LinksProvider.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+
+namespace OpenTelemetry.Trace
+{
+    public sealed class LinksProvider
+    {
```

Comment:
This type is not needed, we'll cache links in span ctor on span class

Reply:
Makes sense. I've opened another draft: #282